### PR TITLE
new class Connection

### PR DESCRIPTION
**Pull request by Dan-Staff**
- Connection to the database is currently a leaky abstraction
- Currently the connection is managed via ugly global variables that are unnatural to document

Solution: Mange database connections through an explicit connection class representing a connection to Fornax.

For example:

```python
url = 'sqlite://'
conn= fornax.Connection(url) as conn:
    print('connected to {}'.format(url))
    graph = fornax.GraphHandle.create(conn)
    graph.add_nodes(label=['a', 'b', 'c'])
print('connection is now closed')
```

alternatively:
```python
url = 'sqlite://'
conn = fornax.Connection(url):
print('connected to {}'.format(url))
graph = fornax.GraphHandle.create(conn)
graph.add_nodes(label=['a', 'b', 'c'])
conn. close()
print('connection is now closed')
```

### CODE REVIEW

File: fornax/api.py [hunk 1]
```diff
@@ -42,65 +42,105 @@
...
+def _set_sqlite_pragma(dbapi_connection, connection_record):
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA foreign_keys=ON")
+    cursor.close()
 
 
-def _hash(item: str) -> int:
-    """An unsalted hash function with a range between 0 and MAX_SIZE
+def _hash(item: str, maxsize=sys.maxsize) -> int:
+    """An unsalted hash function with a range between 0 and MAX_SIZEMAX_SIZE
```
Comment (mattdean-digicatapult) [inline fornax/api.py]: should this now be maxsize?

Comment (Dan-Staff) [reply fornax/api.py]: yes. I've pushed a new commit.

Comment (Dan-Staff) [reply fornax/api.py]: I'll update the other docstrings too...

Comment (Dan-Staff) [reply fornax/api.py]: done

**Pull request merged**